### ISSUE TITLE
Restore cache functionality

### ISSUE THREAD

**Issue by vifactor** [vifactor/repostat#58]
Store fetched data in a easily readable format and next time when repo is about to be analyzed fetch only updated git objects.

**Comment by jonespm** [comment]
Our repository is large and takes almost 5 minutes to fetch the author info. Caching even that would be a nice improvement!

**Comment by vifactor** [comment]
yeah, I filed this issue when I tried to work on a repo where analysis phase lasted more than 20 mins. I have it in plans, but the feature is not something quick. Do not have a deadline, but definitely will prioritize.

**Comment by vifactor** [comment]
History cache might be organized as follows:

```
merge_base = pygit2.Repository.merge_base("cache head", "current head")
```

-  if "cache head" exists in repo, one needs to use all commits "before" "merge_base" from cache and fetch data from
   "merge_base" to a current "head commit"

- if "cache head" is not in repo anymore:
    a) either fetch whole history from scratch
    b) or find a commit in cache that still exists in repo and find "merge_base" with "current_head"
